### ISSUE TITLE
`whitespace` rule is incompatible with `typescript` `<= 2.3.0`

### ISSUE THREAD

**Issue by otaran** [palantir/tslint#2891]
### Bug Report

- __TSLint version__: `5.4.2`
- __TypeScript version__: `2.2.2`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const answer=42;
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "whitespace": true
    }
}
```

#### Actual behavior

`const answer=42` is not flagged by `tslint`, the following warning is reported instead:

```
Warning: ts.isWhiteSpaceLike is not a function
```

#### Expected behavior

`const answer=42` should be flagged by `tslint`with the following message:

```
ERROR: test.ts[1,13]: missing whitespace
ERROR: test.ts[1,14]: missing whitespace
```

#### Notes

`typescript` renamed its internal function `isWhiteSpace` to `isWhiteSpaceLike` in Microsoft/TypeScript@4fa23127fcd9281c39e87120d5f690ec293a7fd8 which was released in `typescript` `2.3.1`.

`tslint` switched to using new name in #2694, but did not provide a fallback code for `typescript` `<= 2.3.0`.

`5.2.0` is the last working version of `tslint`.

**Comment by ajafff** [comment]
`space-before-function-paren` is also affected.

I wondered why the tests didn't fail...
Since #2425 we're no longer testing with older versions of typescript (currently only 2.3.3 and next). Is that intended @adidahiya @nchen63?
I think there should be tests with 2.1.x and 2.2.x

**Comment by adidahiya** [comment]
yeah, looks like that slipped under my radar... I agree that we should be testing with 2.2 and maybe 2.1 (if it's not too inconvenient)
